go/vt/sqlparser/sql.y

```diff
@@ -1304,11 +1304,24 @@ indexes_or_keys:
   }
 
 index_or_key:
-    INDEX
+  INDEX
+  {
+    $$ = string($1)
+  }
+| KEY
+  {
+    $$ = string($1)
+  }
+
+index_or_key_opt:
+  {
+    $$ = ""
+  }
+| INDEX
```

Comment:
Couldn't this be `| index_or_key`?

Reply:
Yeah, i was scared to nest it that way (had problems with grammar conflicts doing similar things in the past) but tried it and it works fine.